### PR TITLE
FEC-GA4 One Login Analytics Implementation

### PR DESCRIPTION
**Pull request by di-aholme**
### What changed

- Updated common express and installed one-login-analytics package
- Updated environment variables
- Modified build scripts and updated relevant config
- Implemented OPL onto each page
- Addition of `edit=true` query parameter (replacing `/edit`) to enable the form change response tracker

### Why did it change

GA4 Implementation

### Issue tracking

- [DFC-318](https://govukverify.atlassian.net/browse/DFC-318)
- [DFC-328](https://govukverify.atlassian.net/browse/DFC-328)
- [DFC-332](https://govukverify.atlassian.net/browse/DFC-332)
- [DFC-396](https://govukverify.atlassian.net/browse/DFC-396)
- [DFC-400](https://govukverify.atlassian.net/browse/DFC-400)

### Environment variables or secrets

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

[DFC-318]: https://govukverify.atlassian.net/browse/DFC-318?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DFC-328]: https://govukverify.atlassian.net/browse/DFC-328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DFC-332]: https://govukverify.atlassian.net/browse/DFC-332?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DFC-396]: https://govukverify.atlassian.net/browse/DFC-396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DFC-400]: https://govukverify.atlassian.net/browse/DFC-400?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/locales/cy/default.yml [hunk 1]
```diff
@@ -63,14 +63,14 @@ error:
     buttonText: "Ewch i hafan GOV.UK"
     buttonLink: "https://www.gov.uk/"
 links:
-  changeAddress: '<a href="/address/edit" id="change-address" data-id="changeAddress" class="govuk-link">Newid<span class="govuk-visually-hidden"> eich cyfeiriad presennol</span></a>'
-  changeYearFrom: '<a href="/address/edit" id="change-address" data-id="changeAddress" class="govuk-link">Newid<span class="govuk-visually-hidden"> y flwyddyn y dechreuoch chi fyw yn eich cyfeiriad presennol</span></a>'
+  changeAddress: '<a href="/address?edit=true" id="change-address" data-id="changeAddress" class="govuk-link">Newid<span class="govuk-visually-hidden"> eich cyfeiriad presennol</span></a>'
```
Comment (di-fabs) [inline src/locales/cy/default.yml:66]: You can't change the path. I suppose it's link to a specific route. 
What we want is to add a new url query param: https://www.npmjs.com/package/@govuk-one-login/one-login-analytics#form-change-tracker
This comment applies to all changes related to the path ".../edit"

File: package.json [hunk 2]
```diff
@@ -9,10 +9,9 @@
...
-    "build-js:application": "mkdir -p dist/public/javascripts; uglifyjs src/assets/javascripts/application.js  node_modules/@govuk-one-login/di-ipv-cri-common-express/src/assets/javascript/application.js --beautify -o dist/public/javascripts/application.js",
-    "build-js:cookies": "mkdir -p dist/public/javascripts; uglifyjs node_modules/@govuk-one-login/di-ipv-cri-common-express/src/assets/javascript/cookies.js --beautify -o dist/public/javascripts/cookies.js",
+    "build-js": "yarn build-js:analytics && yarn build-js:all",
+    "build-js:analytics": "mkdir -p dist/public/javascripts; uglifyjs node_modules/@govuk-one-login/one-login-analytics/lib/analytics.js node_modules/@govuk-one-login/di-ipv-cri-common-express/src/assets/javascript/analytics/ua/legacy/cookies.js --beautify -o dist/public/javascripts/analytics.js",
```
Comment (di-fabs) [inline package.json:13]: ua/legacy/cookie.js is a file which manages the cookie banner. It's now something our analytics package deals with. So we don't want to use it if our package is used.
UA is managed by the script ua/init.js. I think this one will help you to manage UA correctly.

**Review by di-fabs — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (di-aholme) [reply package.json]: Scripts updated to refer to `/ua/init.js`

File: src/locales/cy/default.yml (hunk 1 above)
Comment (di-aholme) [reply src/locales/cy/default.yml]: Reverted the swap, query parameter added as an addition rather than a replacement

**Review by SophiaDWokoma — DISMISSED** [review]

**Review by Caitlin-cooling — APPROVED** [review]
Looks good to me!

**Review by merlinc — COMMENTED** [review]
I've extracted out the "add GA4" into a separate PR - https://github.com/govuk-one-login/ipv-cri-address-front/pull/829 - that PR is now similar to the implementations in the other Orange CRIs

I pushed up the original branch under a different name and [this is a diff between that branch and this one](https://github.com/govuk-one-login/ipv-cri-address-front/pull/822/files) and there's not much difference between the two.

There's a few changes I'd like, but we can fix forward.

**Review by surajkumar — DISMISSED** [review]

**Review by merlinc — CHANGES_REQUESTED** [review]
There might be a Content Security Policy issue that needs to be fixed first

File: src/locales/cy/default.yml [hunk 3]
```diff
@@ -63,14 +63,14 @@ error:
     buttonText: "Ewch i hafan GOV.UK"
     buttonLink: "https://www.gov.uk/"
 links:
-  changeAddress: '<a href="/address/edit" id="change-address" data-id="changeAddress" class="govuk-link">Newid<span class="govuk-visually-hidden"> eich cyfeiriad presennol</span></a>'
-  changeYearFrom: '<a href="/address/edit" id="change-address" data-id="changeAddress" class="govuk-link">Newid<span class="govuk-visually-hidden"> y flwyddyn y dechreuoch chi fyw yn eich cyfeiriad presennol</span></a>'
+  changeAddress: '<a href="/address/edit?edit=true" id="change-address" data-id="changeAddress" class="govuk-link">Newid<span class="govuk-visually-hidden"> eich cyfeiriad presennol</span></a>'
```
[the remark this answers is not in the document]
Comment (di-aholme) [reply src/locales/cy/default.yml:66]: Removed a few changes related to that as the tests were using the pathname as a check, so adding the query parameter there was causing failures

**Review by merlinc — APPROVED** [review]

**Pull request merged**